### PR TITLE
Add an XOR gate

### PR DESCRIPTION
**Pull request by arlyon**
This PR adds the xor operator, implementing it with a simple gate.

This could be extracted into Expr and maybe sped up by using xor directly instead of creating a gate, which doubles the number of evaluations, but since adding new cases could be a breaking change, I decided to leave it.

I also added an example to `Expr` showing the bitwise operators and the `evaluate` method.

This is the reference when writing the xor gate. Maybe I should add a comment.

bilde

### CODE REVIEW

**Comment by arlyon** [comment]
Turns out my 'clever' solution with the xor in the example is just wrong 😢 oops. I'll fix it and push new changes soon.

**Review by cfallin — APPROVED** [review]
LGTM, thanks!

I didn't see an issue with the doc-comment example (is the "river crossing" snippet the one you're referring to?) but I might have missed something. Will wait for confirmation that you've fixed or are OK with it before I merge.

**Comment by arlyon** [comment]
I`ve updated it now, it is modelling the chicken wolf grain problem (https://en.wikipedia.org/wiki/Wolf,_goat_and_cabbage_problem) which I thought I could reduce to use an xor but turns out that that is wrong, which makes me very glad I'm just contributing some qol features and not the expression simplification algorithms! haha

All sorted now!

**Comment by cfallin** [comment]
Thanks, the example definitely improves the docs!

**Pull request merged**
